server.py

```diff
@@ -26,10 +43,26 @@
         statement = fl.read()
     questions[i] = Question(output=output, statement=statement)
 
-
 @app.route("/")
 def changePath():
-    return redirect("/question/1")
+    return redirect("/dashboard")
+
+@app.get("/dashboard")
+def dashboard():
+    return template("dashboard.html", contests=contests)
+
+@app.get("/contest/<code>/<number>")
+def contest(code,number):
+    statement = questions[number].statement
+    return template("index.html", question_number=number, question=statement)
+
+@app.get("/contest/<code>")
+def contest(code):
+    for contest in contests:
```

Comment:
wouldn't it be nice if the contests list was a dictionary indexed by code instead? :thinking:

Reply:
I was thinking the same, just did not implement it earlier for simplicity and forgot. Will do it in a while.